src/dealDashboard/discussionsService.ts

```diff
@@ -232,7 +230,12 @@ export class DiscussionsService {
     if (timestamp)
       this.discussions[discussionId].modifiedAt = timestamp;
 
-    this.dataSourceDeals.update(discussionId, JSON.stringify(this.discussions[discussionId]));
+    this.dataSourceDeals.addClauseDiscussion(
```

Comment:
This is supported in DealTokenSwap.  Would be better to use that.  Note that DealTokenSwap is maintaining clauseDiscussions which I assume we need to keep in sync because the discussions components are using it.